lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3FileSystemConfig.java

```diff
@@ -522,4 +524,38 @@ public S3FileSystemConfig setNonProxyHosts(String nonProxyHosts)
         this.nonProxyHosts = ImmutableSet.copyOf(Splitter.on(',').omitEmptyStrings().trimResults().split(nullToEmpty(nonProxyHosts)));
         return this;
     }
+
+    public Optional<String> getTruststorePath()
+    {
+        return truststorePath;
+    }
+
+    @Config("s3.truststore-path")
+    @ConfigDescription("Path to a Java truststore file for S3 File System Client")
+    public S3FileSystemConfig setTruststorePath(String truststorePath)
+    {
+        this.truststorePath = Optional.ofNullable(truststorePath);
+        return this;
+    }
+
+    public Optional<String> getTruststorePassword()
+    {
+        return truststorePassword;
+    }
+
+    @Config("s3.truststore-password")
+    @ConfigDescription("Password to a Java truststore file for S3 File System Client")
+    @ConfigSecuritySensitive
+    public S3FileSystemConfig setTruststorePassword(String truststorePassword)
+    {
+        this.truststorePassword = Optional.ofNullable(truststorePassword);
+        return this;
+    }
+
+    @AssertTrue(message = "Properties s3.truststore-path and s3.truststore-password should be both set altogether or not set at all")
+    public boolean isTrustStoreConfigValid()
+    {
+        return (truststorePath.isPresent() && truststorePassword.isPresent()) ||
```

Comment:
brackets not needed

Reply:
ok, but I'm not going to change it to !(truststorePath.isEmpty() ^ truststorePassword.isEmpty()) ;)